File: apps/SemanticGraph.Visualizer/src/SemanticGraph.Visualizer.Client/Models/GraphEdgeEvidenceDto.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace SemanticGraph.Visualizer.Client.Models;

public sealed record GraphEdgeEvidenceDto(
    [property: JsonPropertyName("id")] long Id,
    [property: JsonPropertyName("runId")] long RunId,
    [property: JsonPropertyName("provider")] string Provider,
    [property: JsonPropertyName("lspMethod")] string? LspMethod,
    [property: JsonPropertyName("sourceFilePath")] string? SourceFilePath,
    [property: JsonPropertyName("startLine")] long? StartLine,
    [property: JsonPropertyName("startCol")] long? StartCol,
    [property: JsonPropertyName("endLine")] long? EndLine,
    [property: JsonPropertyName("endCol")] long? EndCol,
    [property: JsonPropertyName("rawJson")] JsonElement? RawJson);

File: crates/semantic-graph-visualizer-server/src/dto/graph_edge_evidence_dto.rs
use serde::{Deserialize, Serialize};
use serde_json::Value;

#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct GraphEdgeEvidenceDto {
    pub id: i64,
    pub run_id: i64,
    pub provider: String,
    pub lsp_method: Option<String>,
    pub source_file_path: Option<String>,
    pub start_line: Option<i64>,
    pub start_col: Option<i64>,
    pub end_line: Option<i64>,
    pub end_col: Option<i64>,
    pub raw_json: Option<Value>,
}
